api/src/main/java/org/pmiops/workbench/db/model/UserDataUseAgreement.java

```diff
@@ -0,0 +1,129 @@
+package org.pmiops.workbench.db.model;
+
+import java.sql.Timestamp;
+import java.util.Objects;
+import javax.persistence.Column;
+import javax.persistence.Entity;
+import javax.persistence.GeneratedValue;
+import javax.persistence.GenerationType;
+import javax.persistence.Id;
+import javax.persistence.Table;
+
+@Entity
+@Table(name = "user_data_use_agreement")
+public class UserDataUseAgreement {
```

Comment:
I *think* you can write this in Kotlin now and drop 3/4 of the code. @ericsong do you know if it's supported for Entity classes?

Reply:
Tried it - having trouble figuring out how to make it support autoincrementing primary keys. Doing something like the following, it won't represent the PK as a `Long` instead of a `long` and so I can't null it out

```@Entity
@Table(name = "user_data_use_agreement")
data class UserDataUseAgreement(
        @Id
        @GeneratedValue(strategy = GenerationType.IDENTITY)
        @Column(name = "user_data_use_agreement_id")
        var userDataUseAgreementId: Long,

        @Column(name="user_id")
        var userId: Long,

        @Column(name = "user_given_name")
        var userGivenName: String,

        @Column(name = "user_family_name")
        var userFamilyName: String,

        @Column(name = "user_initials")
        var userInitials: String,

        // This is set to 'true' whenever user_given_name or user_family_name are not the same as the
        // given_name or family_name on the User entry referenced by user_id.
        @Column(name = "user_name_out_of_date")
        var userNameOutOfDate: Boolean,

        @Column(name = "data_use_agreement_signed_version")
        var dataUseAgreementSignedVersion: Int,

        @Column(name = "completion_time")
        var completionTime: Timestamp
)```

Reply:
(this may still be doable, but I don't think it's worth doing now, the day of the deadline)

Reply:
👍 I think we can hold off on using it for now. It's a little tricky with entity models. I'll run a Kotlin primer soon™